common/lib/xmodule/xmodule/js/src/sequence/display.js

```diff
@@ -0,0 +1,409 @@
+            // '#problem_#{id}' -> progress
+            this.progressTable = {};
+        };
+
+        Sequence.prototype.updatePageTitle = function() {
+            // update the page title to include the current section
+            var positionLink = this.link_for(this.position);
+
+            if (positionLink && positionLink.data('page-title')) {
+                document.title = positionLink.data('page-title') + this.base_page_title;
+            }
+        };
+
+        Sequence.prototype.hookUpContentStateChangeEvent = function() {
+            var self = this;
+
+            return $('.problems-wrapper').bind('contentChanged', function(event, problemId, newContentState, newState) {
+                return self.addToUpdatedProblems(problemId, newContentState, newState);
+            });
+        };
+
+        Sequence.prototype.addToUpdatedProblems = function(problemId, newContentState, newState) {
+            /**
+            * Used to keep updated problem's state temporarily.
+            * params:
+            *   'problem_id' is problem id.
+            *   'new_content_state' is the updated content of the problem.
+            *   'new_state' is the updated state of the problem.
+            * initialize for the current sequence if there isn't any updated problem
+            * for this position.
+            */
+
+            if (!this.anyUpdatedProblems(this.position)) {
+                this.updatedProblems[this.position] = {};
+            }
+
+            // Now, put problem content and score against problem id for current active sequence.
+
+            this.updatedProblems[this.position][problemId] = [newContentState, newState];
+        };
+
+        Sequence.prototype.anyUpdatedProblems = function(position) {
+            /**
+            * check for the updated problems for given sequence position.
+            * params:
+            *   'position' can be any sequence position.
+            */
+            return this.updatedProblems[position] !== void 0;
+        };
+
+        Sequence.prototype.hookUpProgressEvent = function() {
+            return $('.problems-wrapper').bind('progressChanged', this.updateProgress);
+        };
+
+        Sequence.prototype.mergeProgress = function(p1, p2) {
+            // if either is 'NA', return the other one
+            var w1, w2;
+            if (p1 === 'NA') {
+                return p2;
+            }
+            if (p2 === 'NA') {
+                return p1;
+            }
+
+            // Both real progresses
+            if (p1 === 'done' && p2 === 'done') {
+                return 'done';
+            }
+
+            // not done, so if any progress on either, in_progress
+            w1 = p1 === 'done' || p1 === 'in_progress';
+            w2 = p2 === 'done' || p2 === 'in_progress';
+
+            if (w1 || w2) {
+                return 'in_progress';
+            }
+
+            return 'none';
+        };
+
+        Sequence.prototype.updateProgress = function() {
```

Comment:
It looks like this method and progressTable can be deleted. Nobody ever uses the values stored in progressTable.